src/d/actor/d_a_bst.cpp

```diff
@@ -4,213 +4,2399 @@
  */
 
 #include "d/actor/d_a_bst.h"
+#include "d/actor/d_a_bomb.h"
+#include "d/d_bg_s_lin_chk.h"
+#include "d/d_camera.h"
+#include "d/d_cc_uty.h"
+#include "d/d_meter.h"
+#include "d/d_snap.h"
+#include "d/res/res_bst.h"
+#include "d/d_s_play.h"
+#include "d/d_com_inf_game.h"
+#include "f_op/f_op_camera.h"
+#include "f_op/f_op_msg_mng.h"
 #include "m_Do/m_Do_ext.h"
+#include "m_Do/m_Do_graphic.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
 #include "d/d_cc_d.h"
+#include "JSystem/JUtility/JUTReport.h"
+
+static u8 hio_set;
+static daBst_HIO_c l_HIO;
 
 /* 000000EC-00000110       .text __ct__11daBst_HIO_cFv */
 daBst_HIO_c::daBst_HIO_c() {
-    /* Nonmatching */
+    field_0x4 = -1;
+    field_0x5 = 0;
+    field_0x6 = 0;
 }
 
+static bst_class* boss;
+static bst_class* hand[2];
+
+static fpc_ProcID msg;
+static s8 msg_end;
+
+
+static u16 set_bdl_d[3] = {BST_BDL_BST, BST_BDL_LHAND, BST_BDL_RHAND};
+static u16 set_za_bdl[3] = {BST_BDL_MUNE, BST_BDL_UDEL, BST_BDL_UDER};
+static u16 set_bck_d[3] = {BST_BCK_BST_SLEEP, BST_BCK_LH_SLEEP, BST_BCK_RH_SLEEP};
+static u16 fly_bck_d[] = {BST_BCK_BST_FLY, BST_BCK_LH_FLY, BST_BCK_RH_FLY};
+static u16 damage_bck_d[] = {BST_BCK_BST_FLY, BST_BCK_LH_DAMAGE, BST_BCK_RH_DAMAGE};
+static u16 down_bck_d[] = {BST_BCK_BST_FLY, BST_BCK_LH_DOWN, BST_BCK_RH_DOWN};
+
 /* 00000110-00000174       .text message_set__FP9bst_classUl */
-void message_set(bst_class*, unsigned long) {
-    /* Nonmatching */
+static BOOL message_set(bst_class* i_this, unsigned long msg_no) {
+    if (msg == -1) {
```

Comment:
msg is a fpc_ProcID, so all the -1s for it should use this enum instead
```suggestion
    if (msg == fpcM_ERROR_PROCESS_ID_e) {
```